pontoon/sync/core.py

```diff
@@ -32,17 +33,19 @@ def sync_project(db_project, now):
         changeset.execute()
 
 
-def serial_task(timeout, **celery_args):
+def serial_task(timeout, lock_key="", base=None, **celery_args):
     """
     Decorator ensures that there's only one running task with given task_name.
     Decorated tasks are bound tasks, meaning their first argument is always their Task instance
     :param timeout: time after which lock is released.
+    :param lock_key: allows to define different lock for respective parameters of task.
     """
+    task_class = base or PontoonTask
     def wrapper(func):
-        @shared_task(bind=True, **celery_args)
+        @shared_task(base=task_class, bind=True, **celery_args)
         @wraps(func)
         def wrapped_func(self, *args, **kwargs):
-            lock_name = "serial_task.{}".format(self.name)
+            lock_name = "serial_task.{}[{}]".format(self.name, lock_key.format(*args, self=self, **kwargs))
```

Comment:
Why pass self into format? I guess we _could_ use the task instance if we wanted but we don't.